### DISCUSSION TITLE
Installation trouble on Mac arm64

### DISCUSSION THREAD

**Discussion by leland45** [UniTime/unitime#180] [Q&A] [answered]
Hello,

I followed installation guidelines for Mac users provided on UniTime website, however I got the following error once I log in :

Cannot invoke "org.unitime.commons.hibernate.util.HibernateContext.getSessionFactory()" because "org.unitime.commons.hibernate.util.HibernateUtil.sContext" is null

At startup, UniTime login page displays following errors :

UniTime 4.8.159 failed to start up properly, please check the application log for more details.

ServiceException: Unable to create requested service [org.hibernate.cache.spi.CacheImplementor] due to: HHH025009: Unable to start region factory (at AbstractServiceRegistryImpl.java:276)
CacheException: HHH025009: Unable to start region factory (at InfinispanRegionFactory.java:246)
EmbeddedCacheManagerStartupException: ISPN000659: Component ComponentWrapper{name=org.infinispan.reactive.publisher.impl.PublisherHandler, instance=org.infinispan.reactive.publisher.impl.PublisherHandler@1b8e1a38, status=FAILED} failed to start (at DefaultCacheManager.java:781)
CacheConfigurationException: ISPN000659: Component ComponentWrapper{name=org.infinispan.reactive.publisher.impl.PublisherHandler, instance=org.infinispan.reactive.publisher.impl.PublisherHandler@1b8e1a38, status=FAILED} failed to start (at BasicComponentRegistryImpl.java:584)
UnsupportedOperationException: getSubject is supported only if a security manager is allowed (at Subject.java:347)

Do you have any hints on what's wrong ?
Regards

**Comment by tomas-muller** [comment, marked as answer]
Are you using Java 23? There appears to be an issue with one of our dependencies. There are two possible solutions:
1) Use an older version of Java. OpenJDK 21 is a good choice (such as [Eclipse Temurin OpenJDK 21](https://adoptium.net/temurin/releases/?os=mac&arch=aarch64&version=21)) as it is used in development and testing and offers a long-time support (LTS). OpenJDK 22 should work fine as well.
2) Use jvm parameter  ``-Djava.security.manager=allow``, e.g., by adding it to the JAVA_OPTS variable before running the Tomcat:
`
export JAVA_OPTS="${JAVA_OPTS} -Xmx2g -Djava.awt.headless=true -Djava.security.manager=allow"
`

**Reply by leland45** [reply]
Hi Tomas,
thanks a lot ! The second option worked perfectly, without the need to install another version of Java (I'm using Java 23).
Regards
